src/main/test/surfrdUtils_test/test_surfrdUtils.pf

```diff
@@ -44,21 +44,130 @@ contains
+     real(r8), allocatable :: wt_in_out(:,:)  ! used in subr. call
+     real(r8) :: expctd(9)
+
+     ! Set relevant pftcon values to defaults; override where necessary
+     call pftcon%InitForTesting()
+     allocate( wt_lunit(begg:endg,istsoil:max_lunit) )
+     allocate( wt_expected(begg:endg,istsoil:max_lunit) )
+     allocate( wt_in_out(begg:endg,istsoil:max_lunit) )
+
+     ! INPUT VALUES
+     wt_lunit(begg:,:) = 0._r8  ! initialize
+     wt_lunit(begg:,1) = (/ 15._r8, 11._r8, 15._r8/)  ! landunit 1
+     wt_lunit(begg:,2) = (/  5._r8,  5._r8,  5._r8/)  ! ...
+     wt_lunit(begg:,4) = (/ 10._r8, 10._r8, 35._r8/)
+     wt_lunit(begg:,5) = (/ 40._r8, 30._r8, 10._r8/)
+     wt_lunit(begg:,6) = (/ 30._r8, 40._r8,  0._r8/)  ! ...
+     wt_lunit(begg:,7) = (/  0._r8,  4._r8, 35._r8/)  ! landunit 7
+     wt_lunit(:,:) = wt_lunit(:,:) / 100._r8
+     call check_sums_equal_1( wt_lunit, begg, "test_check_sums_add_to_1", &
+                              "should not trigger an error")
+
+     do test = 1, tests
+
+        ! OUTPUT VALUES EXPECTED
+        if (test == 1 .or. test == 5) then
+           wt_expected = wt_lunit
+        else if (test == 2) then  ! n_dom_landunits = 1
+           expctd(1) = 40._r8 / 40._r8
+           expctd(2) = 35._r8 / 35._r8
+           wt_expected(begg:,:) = 0._r8  ! initialize
+           wt_expected(begg:,4) = (/ 0._r8, 0._r8, expctd(2) /)  ! landunit 4
+           wt_expected(begg:,5) = (/ expctd(1), 0._r8, 0._r8 /)  ! landunit 5
+           wt_expected(begg:,6) = (/ 0._r8, expctd(1), 0._r8 /)  ! landunit 6
+        else if (test == 3) then  ! n_dom_landunits = 2
+           expctd(1) = 40._r8 / 70._r8
+           expctd(2) = 30._r8 / 70._r8
+           expctd(3) = 35._r8 / 70._r8
+           wt_expected(begg:,:) = 0._r8  ! initialize
+           wt_expected(begg:,4) = (/ 0._r8, 0._r8, expctd(3) /)  ! landunit 4
+           wt_expected(begg:,5) = (/ expctd(1), expctd(2), 0._r8 /)  ! ...
+           wt_expected(begg:,6) = (/ expctd(2), expctd(1), 0._r8 /)  ! ...
+           wt_expected(begg:,7) = (/ 0._r8, 0._r8, expctd(3) /)  ! landunit 7
+        else if (test == 4) then  ! n_dom_landunits = 3
+           expctd(1) = 40._r8 / 85._r8
+           expctd(2) = 30._r8 / 85._r8
+           expctd(3) = 15._r8 / 85._r8
+           expctd(4) = 40._r8 / 81._r8
+           expctd(5) = 30._r8 / 81._r8
+           expctd(6) = 11._r8 / 81._r8
+           expctd(7) = 35._r8 / 85._r8
+           expctd(8) = 35._r8 / 85._r8
+           expctd(9) = 15._r8 / 85._r8
+           wt_expected(begg:,:) = 0._r8  ! initialize
+           wt_expected(begg:,1) = (/ expctd(3), expctd(6), expctd(9) /)  ! ... 1
+           wt_expected(begg:,4) = (/ 0._r8, 0._r8, expctd(8) /)  ! landunit 4
+           wt_expected(begg:,5) = (/ expctd(1), expctd(5), 0._r8 /)  ! ... 5
+           wt_expected(begg:,6) = (/ expctd(2), expctd(4), 0._r8 /)  ! ... 6
+           wt_expected(begg:,7) = (/ 0._r8, 0._r8, expctd(7) /)  ! landunit 7
+        end if
```

Comment:
The main comment I have here is to think about how to make it more obvious what the expected values should be. With some work I can go through and make sure they are right, but, I do have to do some addition in my head. I don't have a concrete suggestion right now, but that's something that I'd like to see. A good goal for a unit test is easy verification that it's correct by looking at it. I know it's harder to do in practice, and I'm sure I've failed miserably at times, but it is good to think about.

Reply:
Ok, I tried something and will push in a minute.

Reply:
I'd say this is better.